rust/rope/src/spans.rs

```diff
@@ -351,8 +364,40 @@ impl<'a, T: Clone> Iterator for SpanIter<'a, T> {
 #[cfg(test)]
 mod tests {
     use super::*;
-    #[test]
 
+    // generate num_spans spans with a pseudo-random composition, values from 0-134456
+    // for a given num_spans it will always generate the same spans, making tests repeatable
+    fn gen_spans(num_spans: usize) -> Node<SpansInfo<usize>> {
```

Comment:
so thinking about this more:

With benchmarks, it's important that we're fully reproducible, but for testing there really is a benefit to randomness, although it is still important to reproduce failures.

I'm particularly interested in some randomness here because of known issues with `Spans`.

With this in mind I'd encourage using `rand` for these cases, but printing out the seed before each test; if we have a failure we can reproduce from the seed.

Reply:
Good idea, I'll use the `rand_pcg` crate for that.